File: slides/nimrodbg/src/RubyBindSample.nim
import ruby

let characterClass = rubyGlobal("Character")
var c = characterClass.new("Bozo")

type Weapon = ref object
  name: string
  damage, durability: int

c.equip(Weapon(name: "Holy Sword of Ultimate Imbalance",
               damage: 99999999,
               durability: 666666666))

c.attack("Kiro")

File: slides/nimrodbg/src/RubyBindSample.rb
class Character
  attr_reader :name, :health

  def initialize(@name)
    @health = 100
  end

  def equip(@weapon)
    puts "#{@name}: I will slash my enemies with my #{@weapon.name}"
  end

  def attack(name)
    puts "#{@name}: Eat my #{@weapon.name}, #{name}"
    enemy = find_character(name)
    enemy.take_damage(@weapon.damage)
  end

  def take_damage(damage)
    @health -= damage
    puts "#{@name}: Aaaarghhhhh"
  end
end
